services/manifest-repo-export-service/pkg/service/version_test.go

```diff
@@ -276,3 +283,281 @@ func TestVersion(t *testing.T) {
+		req     *api.GetManifestsRequest
+		want    *api.GetManifestsResponse
+		wantErr error
+	}
+
+	appName := "app-default"
+	appNameOther := "app-other"
+	fixtureRequest := func(mods ...func(*api.GetManifestsRequest)) *api.GetManifestsRequest {
+		req := &api.GetManifestsRequest{
+			Application: appName,
+			Release:     "latest",
+		}
+		for _, m := range mods {
+			m(req)
+		}
+		return req
+	}
+	fixtureSetupEnv := func() []repository.Transformer {
+		return []repository.Transformer{
+			&repository.CreateEnvironment{
+				Environment: "development",
+				Config: config.EnvironmentConfig{
+					Upstream: &config.EnvironmentConfigUpstream{
+						Latest: true,
+					},
+				},
+				TransformerMetadata: repository.TransformerMetadata{AuthorName: "author-name", AuthorEmail: "author-email"},
+			},
+			&repository.CreateEnvironment{
+				Environment: "staging",
+				Config: config.EnvironmentConfig{
+					Upstream: &config.EnvironmentConfigUpstream{
+						Latest: true,
+					},
+				},
+				TransformerMetadata: repository.TransformerMetadata{AuthorName: "author-name", AuthorEmail: "author-email"},
+			},
+		}
+	}
+	fixtureRelease := func(application string, release uint64) *repository.CreateApplicationVersion {
+		return &repository.CreateApplicationVersion{
+			Authentication:  repository.Authentication{},
+			Application:     application,
+			Version:         release,
+			Team:            "team-123",
+			WriteCommitData: false,
+			Manifests: map[string]string{
+				"development": fmt.Sprintf("dev-manifest for %s in release %d", application, release),
+				"staging":     fmt.Sprintf("staging-manifest for %s in release %d", application, release),
+			},
+			SourceCommitId: fmt.Sprintf("%x",
+				md5.Sum(
+					[]byte(
+						fmt.Sprintf("source commit id for app %s and release %d", application, release),
+					),
+				),
+			),
+			TransformerMetadata: repository.TransformerMetadata{AuthorName: "author-name", AuthorEmail: "author-email"},
+		}
+	}
+
+	fixtureReleaseToManifests := func(release *repository.CreateApplicationVersion) *api.GetManifestsResponse {
+		return &api.GetManifestsResponse{
+			Release: &api.Release{
+				Version:        release.Version,
+				SourceCommitId: release.SourceCommitId,
+			},
+			Manifests: map[string]*api.Manifest{
+				"development": {
+					Environment: "development",
+					Content:     release.Manifests["development"],
+				},
+				"staging": {
+					Environment: "staging",
+					Content:     release.Manifests["staging"],
+				},
+			},
+		}
+	}
+
+	for _, tc := range []*testCase{
+		func() *testCase {
+			release := fixtureRelease(appName, 30)
+
+			return &testCase{
+				name: "happy path",
+				setup: []repository.Transformer{
+					fixtureRelease(appNameOther, 10),
+					fixtureRelease(appNameOther, 20),
+					fixtureRelease(appName, 10),
+					fixtureRelease(appName, 20),
+					release,
+				},
+				req:  fixtureRequest(),
+				want: fixtureReleaseToManifests(release),
+			}
+		}(),
```

Comment:
I think this would be a bit easier to read, if you pull up the line `release := fixtureRelease(appName, 30)` above the for loop. Then the array element does not need to be an inline function.